swarm/api/api.go

```diff
@@ -202,3 +212,151 @@ func (self *Api) Modify(uri, contentHash, contentType string, nameresolver bool)
 	}
 	return trie.hash.String(), nil
 }
+
+func (self *Api) AddFile(mhash, path, fname string, content []byte, nameresolver bool) (storage.Key, string, error) {
```

Comment:
@jmozah #3773 got merged so at the very least you will have to use `api.Parse` and `api.Resolve` in place of `api.parseAndResolve` (I already made a similar change in `swarmfs_unix.go` on master [here](https://github.com/ethereum/go-ethereum/commit/71fdaa42386173da7bfa13f1728c394aeeb4eb01#diff-15577f71e7fabf5288e19e2da204c411)).